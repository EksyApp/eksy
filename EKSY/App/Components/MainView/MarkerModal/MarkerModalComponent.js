import React, {Component} from 'react'
import {Modal, TouchableWithoutFeedback} from 'react-native'
import MarkerModalView from "./MarkerModalView";
import {Actions} from 'react-native-router-flux'
import PropTypes from 'prop-types'
import {MarkerShape, UserShape} from "../../../Utils/PropTypeShapes";

export default class MarkerModalComponent extends Component {
	
	render() {
		return (
				<TouchableWithoutFeedback onPress={this.props.setMarkerViewHidden}>
					<Modal
							visible={this.props.markerViewVisible}
							animationType={'fade'}
							transparent
							onRequestClose={this.props.setMarkerViewHidden}
					>
						<MarkerModalView
								marker = {this.props.marker}
								user = {this.props.user}
						    onEditClick={() => {
						    	Actions.editMarker()
							    this.props.setMarkerViewHidden()
						    }}
								onAddClick={() => {
									Actions.addToRoute()
									this.props.setMarkerViewHidden()
								}}
								routeIsActive={this.props.routeIsActive}
						/>
					</Modal>
				</TouchableWithoutFeedback>
		)
	}
}

MarkerModalComponent.propTypes = {
	setMarkerViewHidden: PropTypes.func,
	markerViewVisible: PropTypes.bool,
	marker: MarkerShape,
	user: UserShape,
	routeIsActive: PropTypes.bool
}