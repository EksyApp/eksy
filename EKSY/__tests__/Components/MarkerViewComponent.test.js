import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MarkerModalComponent from "../../App/Components/MainView/MarkerModal/MarkerModalComponent";

test('MarkerModalComponent renders correctly', () => {
	const tree = renderer.create(
			<MarkerModalComponent
					marker={{
						title: "Hieno marker",
						text: "Hieno tarina",
					}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});