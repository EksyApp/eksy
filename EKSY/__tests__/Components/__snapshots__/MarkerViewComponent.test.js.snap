// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerViewComponent renders correctly 1`] = `
<Modal
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  animationType="fade"
  hardwareAccelerated={false}
  hitSlop={undefined}
  onLayout={undefined}
  onRequestClose={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={undefined}
  testID={undefined}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(243,244,229)",
        "borderRadius": 20,
        "flex": 1,
        "justifyContent": "center",
        "margin": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "margin": 10,
          },
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(101,114,122)",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {
              "color": "rgb(101,114,122)",
              "fontSize": 22,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Hieno marker
      </Text>
      <RCTScrollView>
        <View>
          <View
            onLayout={[Function]}
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <Text
              accessible={true}
              afterCollapse={[Function]}
              afterExpand={[Function]}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={null}
              style={
                Array [
                  Object {
                    "color": "rgb(89,85,78)",
                    "fontSize": 18,
                    "margin": 10,
                  },
                  undefined,
                ]
              }
            >
              Hieno tarina
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
